src/coreclr/src/jit/optimizer.cpp

```diff
@@ -8659,6 +8659,145 @@ bool Compiler::optIsRangeCheckRemovable(GenTree* tree)
     return true;
 }
 
+//----------------------------------------------------------------------------------
+// optBranchlessConditions: optimize simple expressions to branchless expressions
+//
+//  Operation:
+//      Look for BBJ_COND basic blocks which target two BBJ_RETURN blocks
+//      and convert them into single branchless expressions, e.g.
```

Comment:
>expressions [](start = 48, length = 11)

nit: May be worth adding a comment that this opt currently only supports when the returned values are of difference of 1.